### DISCUSSION TITLE
make neo-tree follow the focused file?

### DISCUSSION THREAD

**Discussion by Raagh** [LazyVim/LazyVim#2681] [Q&A] [answered]
Hey so I have this issue, I want neo-tree to focus the file I have focused in my main buffer when I toggle it.

This is my config 

```lua
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      {
        "<leader>e",
        function()
          require("neo-tree.command").execute({ toggle = true, dir = vim.loop.cwd() })
        end,
        desc = "Explorer",
      },
      {
        "<leader>E",
        false,
      },
      { "<leader>fe", false },
      { "<leader>fE", false },
    },
    opts = {
      filesystem = {
        follow_current_file = { enabled = true },
      },
      window = {
        position = "float",
      },
    },
  },

```
As you can see it doesnt focus the file I have in my buffer behind it

![image](https://github.com/LazyVim/LazyVim/assets/8405459/a2324a36-9979-42ae-afc5-83e0616f5253)

**Comment by alewis001** [comment]
The command `:Neotree reveal` I think is the behaviour you're looking for?

**Comment by alewis001** [comment, marked as answer]
If it helps my `lua/plugins/neo-tree.lua` looks like this:
```lua
return {
  "nvim-neo-tree/neo-tree.nvim",
  opts = {
    window = {
      position = "float",
    },
  },
  keys = {
    {
      "<leader>e",
      "<cmd>Neotree reveal<cr>",
      desc = "Explorer Neotree (reveal)",
    },
  },
}
```

**Reply by Raagh** [reply]
Thanks this worked :)

**Comment by dangdd2003** [comment]
i don't know what happen with neotree.nvim, but if you bind keymaps with lua function, the setting position = "float" will break the follow_current_file setting. You should bind with command "Neotree reveal" like @alewis001 

![image](https://github.com/LazyVim/LazyVim/assets/124034919/6ab41a40-fec9-4fa5-aab7-acd27bcca063)
